music-journal/components/AddMemory.js

```diff
@@ -17,21 +28,21 @@ const AddMemory = (props) => {
             <Row className="justify-content-center text-center">
                 <Form>
                     <Row className="justify-content-center">
-                        <Form.Group>
+                        <Form.Group controlId="formSong">
                             <Form.Label>Write a song name</Form.Label>
-                            <Form.Control id="song" as="textarea" rows="1" />
+                            <Form.Control as="textarea" onChange={({target}) => setSong(target.value)} rows="1" />
                         </Form.Group>
                     </Row>
                     <Row>
-                        <Form.Group>
+                        <Form.Group controlId="formMemoryText">
                             <Form.Label>Write something about it</Form.Label>
-                            <Form.Control id="memorytext" as="textarea" rows="3" />
+                            <Form.Control as="textarea" onChange={({target}) => setMemoryText(target.value)} rows="3" />
                         </Form.Group>
                     </Row>
                 </Form>
             </Row>
             <Row className="justify-content-center">
-                <Button onClick={() => addMemory()}>Submit!</Button>
+                <Button onClick={(event) => handleSubmit(event)}>Submit!</Button>
```

Comment:
So the event is the onClick event, and it'll trigger handleSubmit which will read the song and memoryText from state as they're set (updated) every time theres a change to either of the input text areas

Reply:
Yep so each input controls itself and is always up to date when you submit